src/components/containers/TraceRequiredPanel.js

```diff
@@ -12,55 +12,36 @@ class TraceRequiredPanel extends Component {
     const {localize: _} = this.context;
     const {children, ...rest} = this.props;
     let showPanel = true;
-    const emptyPanelMessage = {heading: '', message: ''};
+    let heading = '';
 
-    const noTraceMessage = {
-      heading: _("Looks like there aren't any traces defined yet."),
-      message: _("Go to the 'Create' tab to define traces."),
-    };
+    if (!this.props.visible || !showPanel) {
```

Comment:
Nice refactoring! Please do one more pass here though... you can get rid of `showPanel` I think, and inline the "Looks like there aren't any traces defined yet" heading and so on :)

Reply:
yeah, that was a bit uncooked